addons/mail/static/src/models/thread.js

```diff
@@ -1613,6 +1630,20 @@ registerModel({
                 thread: this,
             });
         },
+        /**
+         * @private
+         */
+        _onChangeThreadViews() {
+            if (
+                this.threadViews.length &&
+                this.messaging.discuss && this.messaging.discuss.discussView
+            ) {
```

Comment:
This condition is not really correct, I think the correct condition should be that current `discuss(view).thread === this`, something like that

But this part of the code might need to be "on change thread in discuss(view)" instead of being here?